projects/telemetry/inc/nmea.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include <stdio.h>
+#include <stdint.h>
+#include "uart_mcu.h"
+typedef enum {
+  GGA = 15, // These are the number of data fields in each message type, excluding the message id
```

Comment:
You probably shouldn't call this `NMEAMessageID` if you're going to be using this to store the number of data fields in each time. If there happens to be another message with the same number of data fields, the `enum` value is no longer unique&mdash;ID implies a unique identifier.

Alternatively, you can do something like

```c
typedef enum {
    NMEA_MESSAGE_ID_RMC = 0,
    NMEA_MESSAGE_ID_GGA,
   // .. and so on,
   NUM_NMEA_MESSAGES
} NMEAMessageID;
```

and use a control table to map to the number of data fields for each data type

```c
uint8_t s_data_fields[NUM_NMEA_MESSAGES] = {
    15,
    8,
    // etc.
};
```

There's also a GCC extension that allows you to specify the index and the value for the array, if you feel that's easier to read. You can take a look at `can_output.c` in driver controls for an example.

Reply:
Got it, we'll change this soon

Reply:
Done